### PR TITLE
Add EEC_DS_TOKEN env var to OTelC StatefulSet.

### PR DESCRIPTION
**Pull request by StefanHauth**
## Description

Add env var `EEC_DS_TOKEN` to OTelC StatefulSet.

## How can this be tested?

1. Deploy Dynakube with extensions
2. check that env var is set on OTelC pods
  ```
  EEC_DS_TOKEN: <set to the key 'otelc.token' in secret 'ext-dk-extensions-token'>  Optional: false 
  ```

### CODE REVIEW

File: pkg/controllers/dynakube/extension/otel/statefulset.go [hunk 1]
```diff
@@ -214,6 +215,13 @@ func buildContainerEnvs(dk *dynakube.DynaKube) []corev1.EnvVar {
 			},
 		},
 		},
+		{Name: envEECDStoken, ValueFrom: &corev1.EnvVarSource{
+			SecretKeyRef: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{Name: dk.Name + consts.SecretSuffix},
```
Comment (luhi-DT) [inline pkg/controllers/dynakube/extension/otel/statefulset.go:220]: dk.Name + consts.SecretSuffix is often used, would it make sense to store it in a combined const instead?

**Review by luhi-DT — CHANGES_REQUESTED** [review]

File: pkg/controllers/dynakube/extension/otel/statefulset.go (hunk 1 above)
Comment (StefanHauth) [reply pkg/controllers/dynakube/extension/otel/statefulset.go:220]: I don't understand. How would you store `dk.Name` in const? It's a variable value. But I can see if it makes sense to put it into a function.

Comment (andriisoldatenko) [reply pkg/controllers/dynakube/extension/otel/statefulset.go:220]: What's @luhi-DT probably wants to create a method of DK instance/type to calculate this string?

Comment (luhi-DT) [reply pkg/controllers/dynakube/extension/otel/statefulset.go:220]: correct

Comment (luhi-DT) [reply pkg/controllers/dynakube/extension/otel/statefulset.go:220]: dk.Name + consts.SecretSuffix is used 3 times, you could create one consts and use it instead, so if the secretname changes you only need to change it in one place

**Review by andriisoldatenko — DISMISSED** [review]

**Review by andriisoldatenko — COMMENTED** [review]
practically LGTM

**Review by aorcholski — APPROVED** [review]
LGTM

**Review by andriisoldatenko — APPROVED** [review]

**Review by luhi-DT — APPROVED** [review]

**Pull request merged**